list_extras.py

```diff
@@ -0,0 +1,42 @@
+"""Print the list of setup/pip extras (options.extras_require section) for a setup.cfg."""
+
+
+import argparse
+import configparser
+from pathlib import Path
+from typing import Iterator
+
+
+def iter_extras(setup_cfg_file: Path) -> Iterator[str]:
+    """Yield each extra key."""
+    cfg = configparser.ConfigParser()
+    cfg.read(setup_cfg_file)
+    try:
+        yield from list(cfg["options.extras_require"].keys())
+    except KeyError:
+        return
+
+
+if __name__ == "__main__":
+
+    def _type_setup_cfg(arg: str) -> Path:
+        fpath = Path(arg)
+        if fpath.name != "setup.cfg":
+            raise ValueError()  # excepted by argparse & formatted nicely
+        if not fpath.exists():
+            raise FileNotFoundError(arg)
```

Comment:
Note that `FileNotFoundError` won't be processed by [argparse](https://docs.python.org/3/library/argparse.html#type) correctly for a help message.
> The argument to type can be any callable that accepts a single string. If the function raises ArgumentTypeError, [TypeError](https://docs.python.org/3/library/exceptions.html#TypeError), or [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError), the exception is caught and a nicely formatted error message is displayed. No other exception types are handled.

Reply:
Wait, does this mean any other Exception is caught and ignored, or just raised to the user verbatim (just not in "a nicely formatted error message")?

Reply:
I believe the latter.  It won't print the help message, just raise a std exception trace.

Reply:
Gotcha. I'm going to use wipac dev tools, after this issue: https://github.com/WIPACrepo/wipac-dev-tools/pull/43